core/src/edu/cornell/gdiac/util/TiledJsonParser.java

```diff
@@ -57,4 +91,395 @@ public static void initPlatformFromJson(
+          obstacle.setBodyType(
+              properties.get("value").asString().equals("static") ? BodyDef.BodyType.StaticBody
+                  : BodyDef.BodyType.DynamicBody);
+          break;
+        case "density":
+          obstacle.setDensity(properties.getFloat("value"));
+          break;
+        case "friction":
+          obstacle.setFriction(properties.getFloat("value"));
+          break;
+        case "restitution":
+          obstacle.setRestitution(properties.getFloat("value"));
+          break;
+        case "debugcolor":
+          try {
+            String cname = properties.getString("value").toUpperCase();
+            Field field = Class.forName(
+                "com.badlogic.gdx.graphics.Color").getField(cname);
+            debugColor = new Color((Color) field.get(null));
+          } catch (Exception e) {
+            debugColor = null;
+          }
+          obstacle.setDebugColor(debugColor);
+          break;
+        case "debugopacity":
+          debugOpacity = properties.getInt("value");
+          obstacle.setDebugColor(obstacle.getDebugColor().mul(debugOpacity / 255.0f));
+          break;
+        case "texture":
+          String key = properties.getString("value");
+          TextureRegion texture = new TextureRegion(directory.getEntry(key, Texture.class));
+          obstacle.setTexture(texture);
+          break;
+        default:
+          break;
+      }
+
+      properties = properties.next();
+    }
+
+    // Call object specific method for object-specific properties
+    switch (json.getString("name")) {
+      case "player":
+        if (obstacle instanceof PlayerModel){
+          initPlayerFromJson((PlayerModel) obstacle, directory, json);
+        }
+        break;
+      case "slope":
+        initSlopeFromJson((SlopeModel) obstacle, json);
+        break;
+      case "bounce":
+        assert (obstacle instanceof BouncePlatformModel);
+        initBounceFromJson((BouncePlatformModel) obstacle, directory, json);
+        break;
+      case "breakable":
+        assert (obstacle instanceof BreakablePlatformModel);
+        initBreakableFromJson((BreakablePlatformModel) obstacle, directory, json);
+        break;
+      case "fan":
+        assert (obstacle instanceof FanModel);
+        initFanFromJson((FanModel) obstacle, directory, json);
+        break;
+      case "exit":
+        setExitDimension((ExitModel) obstacle, directory, json);
+        break;
+      default:
+        break;
+    }
+  }
+
+  /**
+   * @param player    player model to initialize
+   * @param directory asset directory
+   * @param json      json for player object
+   */
+  public static void initPlayerFromJson(
+      PlayerModel player, AssetDirectory directory, JsonValue json) {
+    // Set dimension and frozen texture
+    float width = json.getFloat("width") * (1 / drawScale.x);
+    float height = json.getFloat("height") * (1 / drawScale.y);
+    player.setDimension(width, height);
```

Comment:
I see that not all objects have a width and height, but this is still a common theme among many objects in the game. So we should have a static method for setting the dimension of an object that has width and height.